services/src/main/java/za/co/sfy/lucid/sams/rest/service/SubjectsService.java

```diff
@@ -36,4 +36,19 @@ public SubjectsResponse updateSubjects(SubjectsRequest subjectsRequest) throws L
         return subjectsResponse;
     }
 
+    public SubjectsResponse saveSubjects(SubjectsRequest subjectsRequest) throws LucidSamsExecutionException {
+
+        Subjects subjects = subjectsMapper.subjectsRequestToSubjects(subjectsRequest);
+
+        subjectsResource.save(subjects, subjectsResource);
+
```

Comment:
This save method returns an identifier value that we slap on the Subjects object and return it to our end user. We need to map this value on the object manually. Please do the following :  

Long generatedKey =  subjectsResource.save(subjects, subjectsResource);
subjects.setID((Math.toIntExact(generatedKey));

Apply wherever else applicable